libs/ml/benchmark/ops/ops.cpp

```diff
@@ -153,4 +160,319 @@ BENCHMARK_TEMPLATE(BM_SqrtForward, double, 1024)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_LogForward, double, 2048)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_LogForward, double, 4096)->Unit(benchmark::kMicrosecond);
+
+template <class T, int N>
+void BM_LogBackward(benchmark::State &state)
+{
+  fetch::math::Tensor<T> input({1, N});
+  fetch::math::Tensor<T> error_signal({1, N});
+
+  std::vector<std::reference_wrapper<fetch::math::Tensor<T> const>> inputs;
+  inputs.push_back(input);
+  fetch::ml::ops::Log<fetch::math::Tensor<T>> log1;
+
+  for (auto _ : state)
+  {
+    log1.Backward(inputs, error_signal);
+  }
+}
+
+BENCHMARK_TEMPLATE(BM_LogBackward, double, 2)->Unit(benchmark::kNanosecond);
+BENCHMARK_TEMPLATE(BM_LogBackward, double, 256)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_LogBackward, double, 512)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_LogBackward, double, 1024)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_LogBackward, double, 2048)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_LogBackward, double, 4096)->Unit(benchmark::kMicrosecond);
+
+template <class T, int N>
+void BM_ExpForward(benchmark::State &state)
+{
+  fetch::math::Tensor<T> input({1, N});
+  fetch::math::Tensor<T> output({1, N});
+
+  std::vector<std::reference_wrapper<fetch::math::Tensor<T> const>> inputs;
+  inputs.push_back(input);
+  fetch::ml::ops::Exp<fetch::math::Tensor<T>> exp1;
+
+  for (auto _ : state)
+  {
+    exp1.Forward(inputs, output);
+  }
+}
+
+BENCHMARK_TEMPLATE(BM_ExpForward, double, 2)->Unit(benchmark::kNanosecond);
+BENCHMARK_TEMPLATE(BM_ExpForward, double, 256)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_ExpForward, double, 512)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_ExpForward, double, 1024)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_ExpForward, double, 2048)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_ExpForward, double, 4096)->Unit(benchmark::kMicrosecond);
+
+template <class T, int N>
+void BM_ExpBackward(benchmark::State &state)
+{
+  fetch::math::Tensor<T> input({1, N});
+  fetch::math::Tensor<T> error_signal({1, N});
+
+  std::vector<std::reference_wrapper<fetch::math::Tensor<T> const>> inputs;
+  inputs.push_back(input);
+  fetch::ml::ops::Exp<fetch::math::Tensor<T>> exp1;
+
+  for (auto _ : state)
+  {
+    exp1.Backward(inputs, error_signal);
+  }
+}
+
+BENCHMARK_TEMPLATE(BM_ExpBackward, double, 2)->Unit(benchmark::kNanosecond);
+BENCHMARK_TEMPLATE(BM_ExpBackward, double, 256)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_ExpBackward, double, 512)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_ExpBackward, double, 1024)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_ExpBackward, double, 2048)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_ExpBackward, double, 4096)->Unit(benchmark::kMicrosecond);
+
+template <class T, int N>
+void BM_DivideForward(benchmark::State &state)
+{
+  fetch::math::Tensor<T> input_1({1, N});
```

Comment:
by default tensors are filled with 0s; so this benchmark will compute 0/0
I wonder if that will be faster than performing a real division?

I suggest for all of these ops that you create or fill all the tensors with Tensor<T>::UniformRandom